statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,257 @@
+import numpy as np 
+import pandas as pd
+#from numbers import Number
+
+
+class SurveyStat(object):
+
+    def __init__(self, data, cluster=None, strata=None, prob_weights=None):
+
+        data, self.cluster = self._check_type(data, cluster)
+        data, self.strata = self._check_type(data, strata)
+        data, self.prob_weights = self._check_type(data, prob_weights)
+        
+        self.p = data.shape[1]
+        self.data = np.asarray(data)
+
+    # preprocessing to be used by init
+    def _check_type(self, data, vname):
```

Comment:
The name of this function is not quite right, it is not only "checking" something.  Maybe something like `_prepare_arg`.